pkg/endpoint/events.go

```diff
@@ -233,7 +233,7 @@ func (ev *EndpointPolicyVisibilityEvent) Handle(res chan interface{}) {
 			return
 		}
 		e.getLogger().Debug("creating visibility policy")
-		nvp, err = policy.NewVisibilityPolicy(proxyVisibility)
+		nvp, err = policy.NewVisibilityPolicy(proxyVisibility, e.K8sNamespace, e.K8sPodName)
```

Comment:
Can you check if the k8s metadata is always set? See - https://github.com/cilium/cilium/blob/main/daemon/cmd/endpoint.go#L431-L431.

Reply:
So if pod name and ns are not set in the endpoint I think it's fine to pass them anyway. I will add an empty check for them in `generateL7AllowAllRules` so that we only set the `PolicyLabelDerivedFrom` metadata label and not set empty string name and namespace.